frontend/projects/upgrade/src/app/features/dashboard/segments/pages/segment-details-page/segment-details-page-content/segment-used-by-section-card/segment-used-by-table/segment-used-by-table.component.ts

```diff
@@ -1,10 +1,126 @@
+import { UsedByTableRow, USED_BY_TYPE } from '../../../../../../../../core/segments/store/segments.model';
+import { CommonStatusIndicatorChipComponent } from '../../../../../../../../shared-standalone-component-lib/components';
 
 @Component({
   selector: 'app-segment-used-by-table',
-  imports: [],
   templateUrl: './segment-used-by-table.component.html',
   styleUrl: './segment-used-by-table.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [
+    CommonModule,
+    MatTableModule,
+    MatChipsModule,
+    MatTooltipModule,
+    TranslateModule,
+    RouterModule,
+    SharedModule,
+    MatProgressBarModule,
+    CommonStatusIndicatorChipComponent,
+  ],
 })
-export class SegmentUsedByTableComponent {}
+export class SegmentUsedByTableComponent implements OnInit {
+  displayedColumns: string[] = ['name', 'type', 'status', 'updatedAt'];
+  dataSource$: Observable<UsedByTableRow[]>;
+  isLoading$ = this.segmentsService.isLoadingSegments$;
+  noDataRowText = 'segments.details.used-by.card.no-data-row.text';
+
+  USED_BY_TRANSLATION_KEYS = {
+    NAME: 'Name',
+    TYPE: 'Type',
+    STATUS: 'Status',
+    UPDATED_AT: 'Last Update',
+  };
+
+  experimentSegmentsInclusion$ = this.segmentsService.allExperimentSegmentsInclusion$;
+  experimentSegmentsExclusion$ = this.segmentsService.allExperimentSegmentsExclusion$;
+  featureFlagSegmentsInclusion$ = this.segmentsService.allFeatureFlagSegmentsInclusion$;
+  featureFlagSegmentsExclusion$ = this.segmentsService.allFeatureFlagSegmentsExclusion$;
+  selectedSegment$ = this.segmentsService.selectedSegment$;
+
+  constructor(private segmentsService: SegmentsService) {}
+
+  ngOnInit() {
+    this.dataSource$ = combineLatest([
+      this.selectedSegment$,
+      this.experimentSegmentsInclusion$,
+      this.experimentSegmentsExclusion$,
+      this.featureFlagSegmentsInclusion$,
+      this.featureFlagSegmentsExclusion$,
+    ]).pipe(
+      map(([segment, expInclusions, expExclusions, flagInclusions, flagExclusions]) => {
+        if (!segment) return [];
+
+        // Use Map to prevent duplicates with experimentId or featureFlagId as the key
+        const usedByMap = new Map<string, UsedByTableRow>();
+
+        // Process experiment segments
+        this.processExperimentSegments(expInclusions, segment.id, usedByMap);
+        this.processExperimentSegments(expExclusions, segment.id, usedByMap);
+
+        // Process feature flag segments
+        this.processFeatureFlagSegments(flagInclusions, segment.id, usedByMap);
+        this.processFeatureFlagSegments(flagExclusions, segment.id, usedByMap);
+
+        // Convert Map values to array and sort by updatedAt (newest first)
+        return Array.from(usedByMap.values()).sort(
+          (a, b) => new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime()
+        );
+      })
+    );
+  }
+
+  private processExperimentSegments(segmentData: any[], segmentId: string, resultMap: Map<string, UsedByTableRow>) {
```

Comment:
These functions are almost exactly the same except for a couple pieces of data, can surely be made into one.

Reply:
I’ve addressed all your comments except this one. I believe it’s better to keep the two functions separate, as they accept different data types and separating them improves clarity and readability. That said, if you have a specific idea in mind for how to combine them, feel free to share it.